master/internal/task/allocation.go

```diff
@@ -399,6 +400,7 @@ func (a *allocation) WatchRendezvous(rID sproto.ResourcesID) (RendezvousWatcher,
 
 	if a.rendezvous == nil {
 		a.rendezvous = newRendezvous(a.model.AllocationID, a.resources, rendezvousTimeoutDuration)
```

Comment:
what if we return ```rendezvous``` and a ```close``` ```func()``` in ```newRendezvous```

so that the compiler will bug you to do something about the close function 

while here it is pretty easy to forget to add a closer

Reply:
yah, that is a pretty good idea.